src/tracks/ui/AffordanceHandle.cpp

```diff
@@ -0,0 +1,105 @@
+#include "AffordanceHandle.h"
+
+#include "../../HitTestResult.h"
+#include "../../ProjectAudioIO.h"
+#include "../../RefreshCode.h"
+#include "../../ViewInfo.h"
+#include "../../SelectionState.h"
+#include "../../ProjectSettings.h"
+#include "../../TrackPanelMouseEvent.h"
+#include "../../WaveClip.h"
+#include "../../ProjectHistory.h"
+#include "../../Track.h"
+#include "../../WaveTrack.h"
+#include "../../../images/Cursors.h"
+
+UIHandlePtr AffordanceHandle::HitAnywhere(std::weak_ptr<AffordanceHandle>& holder, const std::shared_ptr<Track>& pTrack)
+    {
+        auto result = std::make_shared<AffordanceHandle>(pTrack);
+        result = AssignUIHandlePtr(holder, result);
+        return result;
+    }
+
+HitTestPreview AffordanceHandle::HitPreview(const AudacityProject* WXUNUSED(pProject), bool unsafe, bool moving)
+{
```

Comment:
Just `const AudacityProject *` if you don't use the argument.  It will make no compiler warning.